pymc_marketing/mmm/budget_optimizer.py

```diff
@@ -112,22 +235,35 @@ def objective_distribution(
         Negative of the total expected contribution for the given budget distribution.
     """
 
-    sum_contributions = 0.0
+    total_response = 0
+    num_days = len(budgets)  # Assuming budgets array corresponds to the days
```

Comment:
Will this work for weekly data? For offline channels, this is often the lowest level of granularity.

Reply:
Yes all granularity types are supported!